pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -42,14 +42,12 @@ func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location str
 			// find the access key with this account
 			key, err := az.getStorageAccesskey(account.Name)
 			if err != nil {
-				glog.V(2).Infof("no key found for storage account %s", account.Name)
-				continue
+				return "", "", fmt.Errorf("no key found for storage account %s", account.Name)
```

Comment:
why return here when there are still other accounts to try?

Reply:
sounds reasonable, I have used `err` var to save the error info, and return `err` in the end, that would be more error clear for user.